extensions/WindowsSandboxExtension/Providers/WindowsSandboxComputeSystem.cs

```diff
@@ -0,0 +1,204 @@
+// Licensed under the MIT License.
+using System.Diagnostics;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Microsoft.Windows.DevHome.SDK;
+using Serilog;
+using Windows.Foundation;
+using Windows.Storage;
+using Windows.Storage.Streams;
+using Windows.Win32;
+using Windows.Win32.Foundation;
+using WindowsSandboxExtension.Helpers;
+using WindowsSandboxExtension.Telemetry;
+
+using Timer = System.Timers.Timer;
+
+namespace WindowsSandboxExtension.Providers;
+
+public class WindowsSandboxComputeSystem : IComputeSystem, IDisposable
+{
+    private readonly Guid _id = Guid.NewGuid();
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(WindowsSandboxProvider));
+    private readonly object _windowsSandboxStartLock = new();
+
+    private Process? _windowsSandboxExeProcess;
+    private ComputeSystemState _state = ComputeSystemState.Stopped;
+
+    private ComputeSystemState State
+    {
+        get => _state;
+
+        set
+        {
+            _state = value;
+            StateChanged?.Invoke(this, value);
+        }
+    }
+
+    public string AssociatedProviderId => Constants.ProviderId;
+
+    public string DisplayName => Resources.GetResource("WindowsSandboxDisplayName", _log);
+
+    public string Id => _id.ToString();
+
+    public string SupplementalDisplayName => string.Empty;
+
+    public ComputeSystemOperations SupportedOperations => ComputeSystemOperations.Terminate;
+
+    public IDeveloperId? AssociatedDeveloperId => null;
+
+    public event TypedEventHandler<IComputeSystem, ComputeSystemState>? StateChanged;
+
+    public IAsyncOperation<ComputeSystemThumbnailResult> GetComputeSystemThumbnailAsync(string options)
+    {
+        return Task.Run(async () =>
+        {
+            var uri = new Uri(Constants.Thumbnail);
+            var storageFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            var randomAccessStream = await storageFile.OpenReadAsync();
+
+            // Convert the stream to a byte array
+            var bytes = new byte[randomAccessStream.Size];
+            await randomAccessStream.ReadAsync(bytes.AsBuffer(), (uint)randomAccessStream.Size, InputStreamOptions.None);
+            return new ComputeSystemThumbnailResult(bytes);
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<IEnumerable<ComputeSystemProperty>> GetComputeSystemPropertiesAsync(string options)
+    {
+        return Task.Run(() =>
+        {
+            var properties = new List<ComputeSystemProperty>
+            {
+                ComputeSystemProperty.Create(ComputeSystemPropertyKind.CpuCount, Environment.ProcessorCount),
+                ComputeSystemProperty.Create(ComputeSystemPropertyKind.AssignedMemorySizeInBytes, 4294967296),
+                ComputeSystemProperty.Create(ComputeSystemPropertyKind.StorageSizeInBytes, 85899345920),
+                ComputeSystemProperty.Create(ComputeSystemPropertyKind.UptimeIn100ns, 100),
```

Comment:
FYI: if you don't provide us with these we won't display them, so its ok to not send these if there is no reliable way to get this data specifically for sandbox

Reply:
I am just providing the defaults for Sandbox. Right now, they are hard coded values, but when we implement the create compute system path, I want users to be able to select different options for this.

Reply:
Oh I see, then I wonder if its a good idea to make these const variable with a name  e.g FourGbInBytes etc so we know at a glance what the value is.